Bug: Screwing up JSON file also screws up AB

Try manipulating your JSON by creating duplicate commissions in your JSON. AB throws you a DuplicateCommissionException exception. We should be handling these exceptions.

> Caused by: seedu.address.model.commission.exceptions.DuplicateCommissionException: Operation would result in duplicate commissions

This bug also appears with iterations (creation of duplicate iterations in JSON throws a DuplicateIterationException exception).
> Caused by: seedu.address.model.iteration.exceptions.DuplicateIterationException: Operation would result in duplicate iterations

FYI: This bug doesn't appear with Customers (duplicate customers in JSON will reset ArtBuddy.json, which is the expected behaviour)

--- a/src/main/java/seedu/address/storage/JsonAdaptedCommission.java
+++ b/src/main/java/seedu/address/storage/JsonAdaptedCommission.java
@@ -19,6 +19,7 @@ import seedu.address.model.commission.Description;
 import seedu.address.model.commission.Fee;
 import seedu.address.model.commission.Title;
 import seedu.address.model.customer.Customer;
+import seedu.address.model.iteration.exceptions.DuplicateIterationException;
 import seedu.address.model.tag.Tag;
 
 /**
@@ -27,6 +28,8 @@ import seedu.address.model.tag.Tag;
 public class JsonAdaptedCommission {
 
     public static final String MISSING_FIELD_MESSAGE_FORMAT = "Commission's %s field is missing!";
+
+    public static final String MESSAGE_DUPLICATE_ITERATIONS = "Iteration list contains duplicate iteration(s).";
 
     private final String title;
     private final String description;
@@ -143,7 +146,11 @@ public class JsonAdaptedCommission {
         modelDescription.ifPresent(commissionBuilder::setDescription);
 
         for (JsonAdaptedIteration iteration : iterations) {
-            commissionBuilder.addIteration(iteration.toModelType());
+            try {
+                commissionBuilder.addIteration(iteration.toModelType());
+            } catch (DuplicateIterationException e) {
+                throw new IllegalValueException(MESSAGE_DUPLICATE_ITERATIONS);
+            }
         }
 
         return commissionBuilder.build(customer);

--- a/src/main/java/seedu/address/storage/JsonAdaptedCustomer.java
+++ b/src/main/java/seedu/address/storage/JsonAdaptedCustomer.java
@@ -11,6 +11,7 @@ import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
 import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.commission.exceptions.DuplicateCommissionException;
 import seedu.address.model.customer.Address;
 import seedu.address.model.customer.Customer;
 import seedu.address.model.customer.Email;
@@ -24,6 +25,8 @@ import seedu.address.model.tag.Tag;
 class JsonAdaptedCustomer {
 
     public static final String MISSING_FIELD_MESSAGE_FORMAT = "Customer's %s field is missing!";
+
+    public static final String MESSAGE_DUPLICATE_COMMISSION = "Commission list contains duplicate commission(s).";
 
     private final String name;
     private final String phone;
@@ -133,7 +136,11 @@ class JsonAdaptedCustomer {
 
         // create commission entries
         for (JsonAdaptedCommission commission : commissions) {
-            customer.addCommission(commission.toModelType(customer));
+            try {
+                customer.addCommission(commission.toModelType(customer));
+            } catch (DuplicateCommissionException e) {
+                throw new IllegalValueException(MESSAGE_DUPLICATE_COMMISSION);
+            }
         }
         return customer;
     }

